cards/portfolio/addon/components/isolated.js

```diff
@@ -52,6 +62,71 @@ export default LiveIsolatedCard.extend({
     return currencyGroups;
   }),
 
+  init() {
+    this._super(arguments);
+
+    if (this.web3.isMetamask) {
+      this.getMetamaskWallet.perform();
```

Comment:
you can `await` this and use an `async init()` if you want to block rendering this component until this task has been performed.